Add content server info to explorer protocol (#37)
decentraland/protocol: docs/Decentraland Protocol for Explorers.md

> A _catalyst node_ is a bundle of services that self-contain a copy of Decentraland. The DAO [governs a curated list of trusted catalysts](https://governance.decentraland.org/?type=catalyst) that hold a working copy of Decentraland. It contains everything necessary to make the Explorer work.

> The content server stores and synchronizes _Entities_. When a new wearable is created, or when a user uploads a scene to their LAND/parcel, all necessary files (3d models, textures, music, etc) are uploaded to a `Content Server`. Each server will verify against the blockchain that the user making the upload is actually allowed to do so. For example, in the case of scenes, the server will check that the deployer owns all the parcels they are trying to modify.

Reviewer: ```suggestion
The content server stores and synchronizes _Entities_. When a new wearable is created, or when a user uploads a scene to their LAND/parcel, all necessary files (3d models, textures, music, etc) are uploaded to a `Content Server`. Each server will verify against the blockchain that the user making the upload is actually allowed to do so. For example, in the case of scenes, the server will check that the deployer address has permissions for all the parcels they are trying to modify.
```

Every time an entity is deployed and accepted by the network, the pointers for that entity will now point to the newest entity for that set of pointers. That is the mechanism used to change the content of the land, wearables and emotes, the entity with the biggest timestamp is the one considered as active.

Unlike LAND, wearables and emotes have [URN pointers](https://github.com/decentraland/urn-resolver/blob/b11aeb677e06e1a9e1d7994efa98a5f11867f854/test/urn.spec.ts#L138-L147). URN are used to reference any asset inside Decentraland, the technology was selected in pursue of a common identifier that enables [extensibility to other networks or remote assets](https://github.com/decentraland/urn-resolver/blob/b11aeb677e06e1a9e1d7994efa98a5f11867f854/test/urn.spec.ts#L269-L280) and [interoperability](https://github.com/common-metaverse/urn-namespaces) with other platforms.

#### Entities format

Entities must be compliant with the defined schema in [Entities Schemas](https://github.com/decentraland/common-schemas/blob/main/src/platform/entity.ts) acording to [ADR-45](https://github.com/decentraland/adr/blob/main/docs/ADR-45-entities-v4.md)

The format of the entities is:

> ```json

Reviewer: ```suggestion
```yaml
```

Author: isn't jsonschema a JSON? 🤔

  timestamp: { type: 'number', minimum: 0 },
  content: { type: 'array', items: ContentMapping.schema },
  metadata: { type: 'object', nullable: true }
}
```

#### Entities Content Restrictions

Each entity type has its own restrictions on the data that is present in the JSON. All the required validations are listed in [ADR-51](https://github.com/decentraland/adr/blob/main/docs/ADR-51-catalyst-content-validations.md), [ADR-62](https://github.com/decentraland/adr/blob/main/docs/ADR-62-merkle-proofed-entities.md), [ADR-74](https://github.com/decentraland/adr/blob/main/docs/ADR-74-add-emote-schema.md).

#### IPFS CIDs

The identifier of the entity is the Hash of its content, the algorithm used is [IPFS with cid version 1](https://github.com/decentraland/hashing).

Every file referenced in the entity is also identified by its hash.

#### Auth Chain

Every entity uploaded to the content server must be signed by the deployer, in order to prove ownership of the pointers that they want to modify.

An auth chain is a chain of signatures, where:

- The first element of the chain is the original signer, and the one who will be used to validate against the blockchain

> - The last element of the chain contains the EHash and a signature

Reviewer: what is an EHash?

> - The last element of the chain contains the EHash and a signature

Author: ```suggestion
- The last element of the chain contains the entity hash and a signature
```

> - In between are ephemeral keys that can be used for signing

Reviewer: ```suggestion
- In between are [ephemeral keys](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt2r1.pdf) that can be used for signing, every ephemeral key is signed with the previous one generating a trust chain. All keys and their signatures conform the AuthChain.
```

The idea is that the user signs an ephemeral key with a certain expiration date. Then that key can be used for signing entities for as long as it remains valid. Therefore, the user only has to sign once until the ephemeral key expires.

In order for the server to validate the authenticity of a deployment, the client will need to send the full chain of signatures. This is an example:

> ```json

Reviewer: ```suggestion
```yaml
```

Global chat (as in chatting with people around in world) will be supported by the comms service, leveraging the P2P protocol which connects people within a single island (or whatever virtual network available). Everyone is allowed to talk in the global chat, even guests.

The [Matrix protocol](https://matrix.org/) will be leveraged for private and channel support, explorers should connect to a Matrix [homeserver](https://matrix.org/faq/#can-i-write-a-matrix-homeserver%3F) and all interactions will be saved in a single Matrix instance (as expected by the protocol), data will not be shared amongst Matrix instances. So, if a given explorer is connected to a Matrix instance, when switching homeservers, the data will not be synchronized between instances. 

On the homeserver side the main responsibility, besides implementing the Matrix protocol, is to support authentication via an AuthChain. This will enable login using a wallet, allowing explorers to use different devices maintaining the same information. Since a wallet is used for login, users will be required to own one for chatting with other users via this method.

> The communications are managed by the Archipielago server in the Catalyst, it's the orchestrator of islands. The islands are the minimal unit of users groups. The universe that a user can see, talk and interact with is delimeted by the island. There will be available [different transports to support this communication: P2P, livekit and websocket](https://github.com/decentraland/adr/blob/main/docs/ADR-70-new-comms.md).

Reviewer: maybe add comms title?

> The communications are managed by the Archipielago server in the Catalyst, it's the orchestrator of islands. The islands are the minimal unit of users groups. The universe that a user can see, talk and interact with is delimeted by the island. There will be available [different transports to support this communication: P2P, livekit and websocket](https://github.com/decentraland/adr/blob/main/docs/ADR-70-new-comms.md).

Author: ```suggestion
#### Communication Service (comms)

The communications are managed by the Archipielago server in the Catalyst, it's the orchestrator of islands. The islands are the minimal unit of users groups. The universe that a user can see, talk and interact with is delimeted by the island. There will be available [different transports to support this communication: P2P, livekit and websocket](https://github.com/decentraland/adr/blob/main/docs/ADR-70-new-comms.md).
```
